docs/design/2020-03-03-SPM-with-parameter-partition.md

```diff
@@ -0,0 +1,109 @@
+
+binding type   |   Priority    | way to create             | action scope
+---------------|:------------- |:--------------------------|-------------------------------------------------
+SQL bind       | first choice  | use SQL                   | one standardized SQL
+Parameter bind | second choice | enable SPM plan evolution | one standardized SQL with some parameters' range
+
+The parameters' range is the range of the value of the constant which in "single column rangeable" condition.
+
+## Compatibility
+
+Incompatible with older versions of **SQL Bind**.
+
+## Implementation
+- Create a new table for parameter bind bucket.
+```sql
+CREATE TABLE IF NOT EXISTS mysql.bind_info_parameter_bucket (
+        original_sql text NOT NULL,
+        bucket_id bigint(64) NOT NULL,
+        count bigint(64) NOT NULL,
+        lower_bound blob default NULL COMMENT "used to Parameter bind, similar with histogram bucket lower bound, composed of a series of constants",
+        upper_bound blob NOT NULL COMMENT "used to Parameter bind, similar with histogram bucket upper bound, composed of a series of constants",
+        unique key bucket(original_sql(1024), bucket_id)
+)
+```
+
+- Add three columns to `mysql.bind_info`
+    1. bind type (options are "SQL bind" and "Parameter bind")
+    2. bucket_id (used to Parameter bind, which bucket belong to)
+    3. execute_time (only updated by SPM plan evolution)
+
+```sql
+CREATE TABLE IF NOT EXISTS mysql.bind_info (
+		original_sql text NOT NULL,
+      	bind_sql text NOT NULL,
+      	default_db text  NOT NULL,
+		status text NOT NULL,
+		create_time timestamp(3) NOT NULL,
+		update_time timestamp(3) NOT NULL,
+		charset text NOT NULL,
+		collation text NOT NULL,
+        bind_type tinyint NOT NULL COMMENT "the type of the bind, options are 0(SQL bind) and 1(Parameter bind)",
+        bucket_id bigint(64) default NULL,
+        execute_time float(64) default NULL,
+		INDEX sql_index(original_sql(1024), default_db(1024)) COMMENT "accelerate the speed when add global binding query",
+		INDEX time_index(update_time) COMMENT "accelerate the speed when querying with last update time"
+	) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;
+```
+
+- Add another parameter binding cache in TiDB
+
+- Change the binding chosen logic
+    1. Reserve the origin SQL bind chosen logic.
+    2. If hits none SQL bind, the try to hit parameter bind.
+
+- SPM plan evolution
+    1. Only change the parameter bind, not affect SQL bind.
+    2. Insert into the bucket like insert histogram bucket in statistics.
+    3. Try to split the bucket if its count exceeds twice count of the smallest bucket.
+    4. If the number of bind SQL for one bucket is more than 20 or another number, drop the binding with max execute_time.
```

Comment:
`execute_time` is for a specific value of parameter, the binding may have larger time for this parameter value, but have smaller time for another parameter value?